iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -774,75 +843,57 @@ static Value getReverseOfCastOp(OpBuilder &b, tensor::CastOp castOp,
   return b.create<memref::CastOp>(castOp.getLoc(), memrefType, resultBuffer);
 }
 
-/// Does a walk of the uses to get to a `flow.dispatch.tensor.store`
-/// operation. If it succeeds returns the store instruction. If not returns
-/// nullptr. Also populates the list of values traversed.
-static Operation *walkUseToStoreOp(Value value,
-                                   SmallVectorImpl<Value> &traversedValues) {
+/// Returns a tied result value give the operand. If no such result exists,
+/// returns `nullptr`.
+static Value getTiedResultForOperand(OpOperand &operand,
+                                     const BufferizationPlan &plan) {
+  for (Value result : operand.getOwner()->getResults()) {
+    if (plan.isEquivalent(operand.get(), result)) {
+      return result;
+    }
+  }
+  return nullptr;
+}
+
+/// To perform updates directly into the result buffer, the uses need to be
+/// walked to get to a value already mapped to a buffer or a
+/// `flow.dispatch.tensor.store` operation. For each use, gets the tied result
+/// and follow its uses. The traversed uses and thir tied results are returned
+/// in `traversedUses`.
+static Value walkUseToGetResultBuffer(
+    OpBuilder &b, Value value, const BufferizationPlan &plan,
+    const BlockAndValueMapping &bvm,
+    SmallVectorImpl<std::pair<OpOperand *, Value>> &traversedUses) {
   Operation *user = nullptr;
   while (value.hasOneUse()) {
     OpOperand &use = *value.use_begin();
     user = use.getOwner();
-    // If the user is a store op, we are done.
-    if (isa<IREE::Flow::DispatchTensorStoreOp>(user)) break;
-    if (auto subTensorInsertOp = dyn_cast<SubTensorInsertOp>(user)) {
-      if (subTensorInsertOp.source() == use.get()) break;
+    // Check if this is a `flow.dispatch.tensor.store` op.
```

Comment:
This is a meaningless comment to me, delete it?

https://google.github.io/styleguide/cppguide.html#Implementation_Comment_Donts